### ISSUE TITLE
[BUG/ISSUE]Problem about applying regional mask to BC emission and updating geoschem to 12.9.3

### ISSUE THREAD

**Issue by hym0627** [geoschem/geos-chem#461]
I have encountered two issues.

The first one:
I would like to apply a regional mask to the black carbon emission. I thought that I could do this by adding a Mask identification number in HEMCO_Config.rc directly,  and it seems that the mask does work but it hasn't worked correctly. 
According to the HEMCO_diagnostics output, the total emission of black carbon decreases after applying regional mask (from 9524  to 8999), but the result is not consistent with the regional emission(2256) that I want to identify.
I would appreciate any help that you could provide. I have pasted my attempt below. I am using GEOS-Chem 12.6.0, Offline Aerosols chemistry, MERRA2 4x5.
1.
In HEMCO_Config.rc, I added a new Mask identification number 1100 to all of the active BC emission code. The rest of the line is unchanged. 
`0 BCPI_COAL    $ROOT/PKU/BC_Tao/$YYYY/coal_$YYYY.nc   BC  1750-2014/1-12/1/0  C  xy  kg/m2/s  BCPI  101/1100  1/2/12 6`

2.Then I added a Mask identification number 1100 to the mask section of HEMCO_Configr.rc . 
`1100 TAGBC_AFR_MASK    $ROOT/MASKS/taggedBC_1x1_mask.nc      taggedBC_AFR   2013/1/1/0  C  xy  1  1  -17.5/-36/52.5/16`

The second one:
As I encountered the issue above , I checked the Github and found a similar issue on [https://github.com/geoschem/geos-chem/issues/221](https://github.com/geoschem/geos-chem/issues/221). The question was furtherly discussed on [https://github.com/geoschem/geos-chem/issues/232](https://github.com/geoschem/geos-chem/issues/232). So I decided to update GEOS-Chem to 12.9.3 to find if this works out. Although geoschem could be compiled successfully with command 'make -j4 build' , the model ran into problems from the very beginning.
1.
I redownloaded source code and unit tester from Github and regenerate a new run directory for GC12.9.3
2.
HEMCO_Config.rc, HEMCO_Diagn.rc HEMCO_Diag.rc were configured exactly the same as in version 12.6.0 I am currently use.
The rest of the files are unchanged.
3.
log file ends like `READ_INPUT_FILE: Opening ./input.geos` without any error message.

### Input and log files to attach
[for question 1.zip](https://github.com/geoschem/geos-chem/files/5291565/for.question.1.zip)
[for question 2.zip](https://github.com/geoschem/geos-chem/files/5291599/for.question.2.zip)

(NOTE: edited by lizziel to enable the referenced github issue links to work)

**Comment by lizziel** [comment]
Hi @hym0627,

Regarding **question 2:**
It looks like you do not have a 12.9.3 run directory. Per the SLURM log file, the run is crashing in `input_mod.F90` in the section for the "Debug output" entry in `input.geos`. Looking at the top of your config file, here is what is there:
```
------------------------+------------------------------------------------------
%%% SIMULATION MENU %%% :
Start YYYYMMDD, hhmmss  : 20140101 000000
End   YYYYMMDD, hhmmss  : 20140102 000000
Run directory           : ./
Root data directory     : /vol-th/home/hym/ExtData
Met field               : merra2
Simulation name         : aerosol
------------------------+------------------------------------------------------
```

When I checkout the 12.9.3 tag for the unit tester, however, I see something different for the 4x5 aerosol `input.geos `template. That file contains this:
```
------------------------+------------------------------------------------------
%%% SIMULATION MENU %%% :
Start YYYYMMDD, hhmmss  : {DATE1} {TIME1}
End   YYYYMMDD, hhmmss  : {DATE2} {TIME2}
Run directory           : ./
Root data directory     : {DATA_ROOT}
Met field               : {MET}
Simulation name         : {SIM}
Species database file   : ./species_database.yml
Turn on debug printout? : F
Use GC classic timers?  : F
------------------------+------------------------------------------------------
```

Check your unit tester version with `git log` to make sure you have the correct version.

That being said, you will run into problems with the simulation even with the correct run directory. There is a known issue that the aerosol-only simulation does not work in 12.9.0-12.9.3. See issue github https://github.com/geoschem/geos-chem/issues/354.

**Comment by lizziel** [comment]
Regarding **question 1:**
I am not sure I understand what is going wrong. Could you provide a global plot of the BCPI_COAL diagnostic before and after the application of the mask? Also, what is the 101 scaleID that you are using with the mask? I have not been able to find it in the aerosols simulation HEMCO_Config.rc template for 12.6.0.

**Comment by lizziel** [comment]
Hi @hym0627, were you able to make progress in solving this issue?

**Comment by hym0627** [comment]
Hi, lizziel! Thanks for your reply and suggestion. I'm sorry for my late reply! Frankly speaking, I didn't make any progress. Regarding my second question,  I check the unit_tester version and make sure  that the version was correct. I also ran a standard simulation, but it ran into the same problem.  The 4x5 standard `input.geos` template in my  unit_tester12.9.3 contains this:
`%%% SIMULATION MENU %%% :
Start YYYYMMDD, hhmmss  : {DATE1} {TIME1}
End   YYYYMMDD, hhmmss  : {DATE2} {TIME2}
Run directory           : ./
Root data directory     : {DATA_ROOT}
Met field               : {MET}
Simulation name         : {SIM}
Species database file   : ./species_database.yml
Turn on debug printout? : F
Use GC classic timers?  : F
`But it is really weird that the `input.geos` generated from this template just contains:
`%%% SIMULATION MENU %%% :
Start YYYYMMDD, hhmmss  : 20140101 000000
End   YYYYMMDD, hhmmss  : 20140201 000000
Run directory           : ./
Root data directory     : /vol-th/home/hym/ExtData
Met field               : merra2
Simulation name         : standard`
The attachments  is configuration file `CopyRunDirs.aerosol`  which I used to generate my run directory with command `./gcCopyRunDirs CopyRunDirs.aerosol`  

[CopyRunDirs.txt](https://github.com/geoschem/geos-chem/files/5390519/CopyRunDirs.txt)

**Comment by lizziel** [comment]
It is possible that you are generating your run directory from a clone of `geos-chem-unittest` other than the one in which you are running `gcCopyRunDir`. Unfortunately this is an issue that occurs time to time due to the way `CopyRunDirs` is written. The clone of `geos-chem-unittest` used to create the run directory is set in `CopyRunDirs` with this line:
`UNIT_TEST_ROOT : {HOME}/UT`

Check that the path of the `geos-chem-unittest` you are using to generate the directory is the same as that. If it is not, then you need to edit that path to point to the clone you are in.

**Comment by hym0627** [comment]
Oh,that's the problem! Thank you lizziel! But I have encountered new issue.
This time my `log` file end with this error message:
```
===============================================================================
GEOS-Chem ERROR: GLOBAL_HCl not found in HEMCO data list!
 -> at DO_ISORROPIAII (in module GeosCore/isorropiaII_mod.F90)
===============================================================================
```
The attachments are my configuration and log file
[HEMCO.log.txt](https://github.com/geoschem/geos-chem/files/5398883/HEMCO.log.txt)
[input.geos.txt](https://github.com/geoschem/geos-chem/files/5398884/input.geos.txt)
[log.txt](https://github.com/geoschem/geos-chem/files/5398885/log.txt)
[HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/5398886/HEMCO_Config.rc.txt)

**Comment by lizziel** [comment]
Hi @hym0627, I noticed from your log files that you are using the aerosol simulation. There is a problem in the aerosol simulation for 12.9 that prevents running and this is what you are seeing. See GitHub issue https://github.com/geoschem/geos-chem/issues/354.

**Comment by hym0627** [comment]
Ok, my problem has been solved. Thank you very much for your help！

**Comment by neyranaz** [comment]
@hym0627 Did you end up using an earlier version of GC to perform your aerosol run, or did you fix 12.9.3? 
We did it an Aerosol run with 12.7.0 and it works, but we were figuring out if we could already use a later version.
